src/components/shared/ButtonsLarge/index.tsx

```diff
@@ -0,0 +1,34 @@
+import classnames from 'classnames';
+
+export enum ButtonLargeVariants {
+  Primary = 'primary',
+  Outline = 'outline',
+}
+
+interface ButtonLargeProps {
```

Comment:
Maybe here you can extend from the HTML Button properties so you don't need to redeclare common things like disabled or onClick..
Something like
```suggestion
interface ButtonLargeProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant: string;
  hasIcon: boolean;
}
```

This would allow to use all the props that the html button is expected to receive plus your special props like variant or hasIcon..  For the button content you could use children to pass its text naturally. This will even allow you in the future to pass any valid child (such as icons or stuff) as children and it should work fine!